aws_google_auth/configuration.py

```diff
@@ -0,0 +1,144 @@
+
+    # Write the configuration (and credentials) out to disk. This allows for
+    # regular AWS tooling (aws cli and boto) to use the credentials in the
+    # profile the user specified.
+    def write(self, amazon_object):
+        self.ensure_config_files_exist()
+
+        assert (self.profile is not None), "Can not store config/credentials if the AWS_PROFILE is None."
+
+        # Write to the configuration file
+        config_parser = configparser.RawConfigParser()
+        config_parser.read(self.config_file)
+        if not config_parser.has_section(self.profile):
+            config_parser.add_section(self.profile)
+        config_parser.set(self.profile, 'region', self.region)
+        config_parser.set(self.profile, 'google_config.ask_role', self.ask_role)
+        config_parser.set(self.profile, 'google_config.duration', self.duration)
+        config_parser.set(self.profile, 'google_config.idp_id', self.idp_id)
+        config_parser.set(self.profile, 'google_config.role_arn', self.role_arn)
+        config_parser.set(self.profile, 'google_config.sp_id', self.sp_id)
+        config_parser.set(self.profile, 'google_config.u2f_disabled', self.u2f_disabled)
+        config_parser.set(self.profile, 'google_config.username', self.username)
+        with open(self.config_file, 'w+') as f:
+            config_parser.write(f)
+
+        # Write to the credentials file (only if we have credentials)
+        if amazon_object is not None:
+            credentials_parser = configparser.RawConfigParser()
+            credentials_parser.read(self.credentials_file)
+            if not credentials_parser.has_section(self.profile):
+                credentials_parser.add_section(self.profile)
+            credentials_parser.set(self.profile, 'aws_access_key_id', amazon_object.access_key_id)
+            credentials_parser.set(self.profile, 'aws_secret_access_key', amazon_object.secret_access_key)
+            credentials_parser.set(self.profile, 'aws_security_token', amazon_object.session_token)
+            credentials_parser.set(self.profile, 'aws_session_expiration', amazon_object.expiration.strftime('%Y-%m-%dT%H:%M:%S%z'))
+            credentials_parser.set(self.profile, 'aws_session_token', amazon_object.session_token)
+            with open(self.credentials_file, 'w+') as f:
+                credentials_parser.write(f)
+
+    # Read from the configuration file and override ALL values currently stored
+    # in the configuration object. As this is potentially destructive, it's
+    # important to only run this in the beginning of the object initialization.
+    # We do not read AWS credentials, as this tool's use case is to obtain
+    # them.
+    def read(self, profile):
+        self.ensure_config_files_exist()
+
+        profile = util.Util.default_if_none(profile, self.profile)
+        config_parser = configparser.RawConfigParser()
+        config_parser.read(self.config_file)
+        if config_parser.has_section(profile):
```

Comment:
Looking up the direct profile doesn't work, they should have a `profile` prefix.

Something like:

```
        if profile is None:
            profile_name = "default"
        else:
            profile_name = "profile {}".format(profile)
```

see: https://docs.aws.amazon.com/cli/latest/userguide/cli-multiple-profiles.html